src/app/map/[id]/hooks/useMapViews.ts

```diff
@@ -3,56 +3,40 @@
 import { useMutation, useQueryClient } from "@tanstack/react-query";
 import { use, useCallback, useMemo } from "react";
 import { toast } from "sonner";
-import { MapContext, ViewConfig } from "@/app/map/[id]/context/MapContext";
+import {
+  MapContext,
+  createNewViewConfig,
+} from "@/app/map/[id]/context/MapContext";
 import { useTRPC } from "@/services/trpc/react";
 import { getNewLastPosition } from "../utils";
 import { useMapQuery } from "./useMapQuery";
 import type { View } from "../types";
+import type { MapViewConfig } from "@/server/models/MapView";
 
 export function useMapViews() {
-  const { viewId, mapId, setDirtyViewIds, setViewId } = use(MapContext);
+  const { viewId, mapId, setViewId, setDirtyViewIds } = use(MapContext);
   const trpc = useTRPC();
   const queryClient = useQueryClient();
   const { data: mapData } = useMapQuery(mapId);
 
-  // Get views directly from cache
-  const views = useMemo(() => mapData?.views || [], [mapData?.views]);
+  // Get views directly from cache and ensure they have isTag field
+  const views = useMemo(() => {
+    const rawViews = mapData?.views || [];
+    return rawViews.map((v) => ({
+      ...v,
+      isTag: v.isTag ?? false, // Ensure isTag is always a boolean
```

Comment:
note: this shouldn't be necessary - `view.isTag` should already be a `boolean`, as in the model (`server/models/MapView`) it is defined as `boolean()`, not `boolean().optional()`, and the column is `NOT NULL` in the database